tmt/steps/provision/podman.py

```diff
@@ -41,13 +41,23 @@ def options(cls, how=None):
             click.option(
                 '-p', '--pull', is_flag=True,
                 help='Force pulling a fresh container image.'),
+            click.option(
+                '--reboot-soft', is_flag=True,
+                help='Reboot container nicely.'),
+            click.option(
+                '--reboot-hard', is_flag=True,
+                help='Reboot container by stop and start.'),
             ] + super().options(how)
```

Comment:
I was thinking a bit more about the best way how to provide the reboot functionality from the command line: What about this: Instead of introducing the same/similar reboot options to each of the provision plugins we could introduce a new `run` subcommand which would handle this:

    tmt run reboot
    tmt run reboot --hard
    tmt run reboot --name guest

This would also resolve the implemenation question of how to re-execute the `go()` method when the actual provisioning has been already completed. To me it seems the new subcommand would nicely fit into the same "group" as the `login` subcommand.

 * **steps** ... discover provision prepare execute report finish
 * **selection** ... plans tests
 * **action** ... login reboot

What do you think, @RHEmployee, @FrNecas, @lukaszachy?

Reply:
Interesting idea, makes sense to me. With some extensions, it could theoretically allow something like `tmt run discover provision prepare execute reboot execute` (where the second execute may be differently parametrized than the first one). This may cover some use cases of tmt, right?

Reply:
We definitely want to support providing multiple step configurations on the command line (this functionality is slightly outlined in #68) so in the future something like this should work:

    tmt run prepare --how install --package tree prepare --how ansible --playbook forest.yaml

Respecting the step order and thus allowing something like `execute reboot execute` sounds interesting. We would need to think this through properly. But I believe this should be doable and could be useful.

Reply:
Seems great. Need to take a look at how execute is handled.

Reply:
Ok. Is there some guideline how to add login like subcommand to tmt?
I found definition of login command in:
tmt/steps/__init__.py
however I'm not sure how to create similar static class for reboot purposes and how to pass arguments to subcommand (looks like login is argumentless).
For example:
I need to provide to static Reboot class which kind of reboot is needed.
There is try_running_login() method in Step class but it's still unclear how to route data araound.
Like: If I want something like: reboot --hard then I need:
create static Reboot class for reboot subcommand and then save what kind of reboot is needed (so parse arguments of subcommand via click?).
Then I need to pass data to class which implements reboot itself and then decide which kind of reboot I need to run.

Reply:
Ok. Hope I made some progress.